src/Workspaces/Core/Portable/Workspace/Solution/ChecksumCollection.cs

```diff
@@ -0,0 +1,73 @@
+﻿// Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
+
+using System;
+using System.Collections;
+using System.Collections.Generic;
+using System.Linq;
+
+namespace Microsoft.CodeAnalysis.Serialization
+{
+    /// <summary>
+    /// collection which children is checksum.
+    /// </summary>
+    internal abstract class ChecksumCollection : ChecksumWithChildren, IEnumerable<Checksum>
+    {
+        protected ChecksumCollection(string kind, Checksum[] checksums) : this(kind, (object[])checksums)
+        {
+        }
+
+        protected ChecksumCollection(string kind, object[] checksums) : base(kind, checksums)
+        {
+        }
+
+        public int Count => Children.Count;
+        public Checksum this[int index] => (Checksum)Children[index];
```

Comment:
it can't be generic since it children can be either checksum or checksumWithChildren. checksumCollection is a version of ChecksumWithChildren whose child is all checksum.